src/test/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminAuditControllerTest.java

```diff
@@ -0,0 +1,130 @@
+            "some data after 2", "mosh court 2",
+            LocalDateTime.of(1002, 2, 12, 12,12, 1)
+        )),
+        new Audit(new uk.gov.hmcts.dts.fact.entity.Audit(
+            3, "kupo email 3", new AuditType(), "some data before 3",
+            "some data after 3", "mosh court 3",
+            LocalDateTime.of(1003, 3, 13, 13,13, 1)
+        ))
+    );
+
+    @BeforeAll
+    static void setUp() throws JsonProcessingException {
+        // If we do not include the register module here, the date will be outputted differently to
+        // what we would expect, and will not match the json output from the controller call
+        // based on how the spring framework deserializes it's requests etc
+        OBJECT_MAPPER.registerModule(new JavaTimeModule());
+        OBJECT_MAPPER.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        auditJson = OBJECT_MAPPER.writeValueAsString(AUDIT_LIST);
+    }
+
+    @Test
+    void shouldRetrieveAuditsWithOnlyPageAndSize() throws Exception {
+
+        when(adminAuditService.getAllAuditData(TEST_PAGE, TEST_SIZE, Optional.empty(), Optional.empty(),
+                                               Optional.empty(), Optional.empty())).thenReturn(AUDIT_LIST);
+        mockMvc.perform(get(String.format(BASE_PATH, TEST_PAGE, TEST_SIZE)))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(content().json(auditJson));
+    }
+
+    @Test
+    void shouldRetrieveAuditsWithPageSizeLocationAndEmail() throws Exception {
+
+        when(adminAuditService.getAllAuditData(TEST_PAGE, TEST_SIZE, Optional.of(TEST_LOCATION), Optional.of(TEST_EMAIL),
+                                               Optional.empty(), Optional.empty())).thenReturn(AUDIT_LIST);
+        mockMvc.perform(get(String.format(BASE_PATH_SEARCH_NO_DATE_RANGE, TEST_PAGE,
+                                          TEST_SIZE, TEST_LOCATION, TEST_EMAIL)))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(content().json(auditJson));
+    }
+
+    @Test
+    void shouldRetrieveAuditsWithPageSizeLocationEmailAndDateRange() throws Exception {
+
+        when(adminAuditService.getAllAuditData(TEST_PAGE, TEST_SIZE, Optional.of(TEST_LOCATION),
+                                               Optional.of(TEST_EMAIL), Optional.of(TEST_DATE_FROM),
+                                               Optional.of(TEST_DATE_TO))).thenReturn(AUDIT_LIST);
+
+        mockMvc.perform(get(String.format(BASE_PATH_SEARCH_WITH_DATE_RANGE, TEST_PAGE,
+                                          TEST_SIZE, TEST_LOCATION, TEST_EMAIL,
+                                          TEST_DATE_FROM, TEST_DATE_TO)))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(content().json(auditJson));
```

Comment:
Both tests return the full audit list. Would it test the logic more if we formulated a request that returned a portion of the data?

Reply:
It probably wouldn't make too much of a difference, since it's returning what we tell the mock to return. It might be clearer, but wouldn't test the logic further